lib/mayaUsd/utils/utilSerialization.cpp

```diff
@@ -294,18 +294,62 @@ SdfLayerRefPtr saveAnonymousLayer(
 
     saveLayerWithFormat(anonLayer, filePath, formatArg);
 
-    SdfLayerRefPtr newLayer = SdfLayer::FindOrOpen(filePath);
+    const bool  isSubLayer = (parent._layerParent != nullptr);
+    std::string relativePathAnchor;
+
+    if (savePathAsRelative) {
+        if (isSubLayer) {
+            filePath
+                = UsdMayaUtilFileSystem::getPathRelativeToLayerFile(filePath, parent._layerParent);
+            relativePathAnchor = UsdMayaUtilFileSystem::getLayerFileDir(parent._layerParent);
+        } else {
+            filePath = UsdMayaUtilFileSystem::getPathRelativeToMayaSceneFile(filePath);
+            relativePathAnchor = UsdMayaUtilFileSystem::getMayaSceneFileDir();
+        }
+    }
+
+    // When the filePath was made relative, we need to help FindOrOpen to locate
+    //      the sub-layers when using relative paths. We temporarily chande the
+    //      current directory to the location the file path is relative to.
+    UsdMayaUtilFileSystem::TemporaryCurrentDir tempCurDir(relativePathAnchor);
+    SdfLayerRefPtr                             newLayer = SdfLayer::FindOrOpen(filePath);
+    tempCurDir.restore();
```

Comment:
I see this pattern with TemporaryCurrentDir in many places. It may be worth creating a utility function.